### DISCUSSION TITLE
el-button绑定icon后必须在button必须输入内容icon才正常显示

### DISCUSSION THREAD

**Discussion by shiminzhong** [element-plus/element-plus#13711] [Q&A]
el-button绑定icon后必须在button必须输入内容icon才正常显示，没有内容就不显示。但是有占位

**Comment by ryuhangyeong** [comment]
If you provide a minimal example, it will help us verify or solve the problem!

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3QgbXNnID0gcmVmKCdIZWxsbyBXb3JsZCEnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGgxPnt7IG1zZyB9fTwvaDE+XG4gIDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgLz5cblxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBlcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

**Comment by shiminzhong** [comment]
https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG4vLyB2dWUz5qGG5p625o+Q5L6b5LiA5Liq5pa55rOVdXNlQXR0cnPmlrnms5XvvIzlroPlj6/ku6Xojrflj5bnu4Tku7bouqvkuIrnmoTlsZ7mgKfkuI7kuovku7ZcbmltcG9ydCB7RWRpdH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5pbXBvcnQgSGludEJ1dHRvbiBmcm9tIFwiLi9IaW50QnV0dG9uLnZ1ZVwiO1xuXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuPGRpdj5cbiAgPGVsLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIHNpemU9XCJzbWFsbFwiIDppY29uPVwiRWRpdFwiID48L2VsLWJ1dHRvbj5cbjwhLS0gIOiHquWumuS5iee7hOS7ti0tPlxuICA8SGludEJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiIHNpemU9XCJzbWFsbFwiIDppY29uPVwiRWRpdFwiPjwvSGludEJ1dHRvbj5cbjwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlIHNjb3BlZD5cblxuPC9zdHlsZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9IaW50QnV0dG9uLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XHJcbi8v5byV5YWldXNlQXR0cnPmlrnms5XvvJrojrflj5bmoIfnrb7kuIrlsZ7mgKfkuI7kuovku7ZcclxuaW1wb3J0IHt1c2VBdHRyc30gZnJvbSBcInZ1ZVwiOyAvL+i/lOWbnuWvueixoVxyXG5sZXQgJGF0dHJzID0gdXNlQXR0cnMoKTtcclxuY29uc29sZS5sb2coJGF0dHJzKVxyXG48L3NjcmlwdD5cclxuXHJcbjx0ZW1wbGF0ZT5cclxuPGRpdj5cclxuICA8ZWwtYnV0dG9uIDo9XCIkYXR0cnNcIj7lrZDnu4Tku7Y8L2VsLWJ1dHRvbj5cclxuPC9kaXY+XHJcbjwvdGVtcGxhdGU+XHJcblxyXG48c3R5bGUgc2NvcGVkPlxyXG5cclxuPC9zdHlsZT4iLCJfbyI6e319










Element Plus Playground







我自己编译器编译的结果




我很奇怪，，所有的版本都是对应的，但是编译后的结果却对不上，只有在标签加入内容才会显示icon图标。







                       
原始邮件
                       
                     

发件人："류한경"< ***@***.*** &gt;;

发件时间：2023/7/24 18:47

收件人："element-plus/element-plus"< ***@***.*** &gt;;

抄送人："shiminzhong"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

主题：Re: [element-plus/element-plus] el-button绑定icon后必须在button必须输入内容icon才正常显示 (Discussion #13711)




 
If you provide a minimal example, it will help us verify or solve the problem!
 
Element Plus Playground
